kie-server-parent/kie-server-maven-plugin/src/main/java/org/kie/server/gateway/KieServerGateway.java

```diff
@@ -0,0 +1,112 @@
+ */
+public class KieServerGateway {
+
+    public static final Logger LOG = LoggerFactory.getLogger(KieServerGateway.class);
+
+    private final KieServicesConfiguration config;
+    private KieServicesClient client;
+
+    public KieServerGateway(String protocol, String hostname, Integer port, String username, String password, Integer timeout) {
+
+        String serverUrl = new StringBuilder(protocol)
+                .append("://")
+                .append(hostname)
+                .append(":")
+                .append(port)
+                .append("/kie-server/services/rest/server")
+                .toString();
+
+        LOG.info("Server Url {}", serverUrl);
+
+        config = KieServicesFactory.newRestConfiguration(serverUrl, username, password);
+        config.setMarshallingFormat(MarshallingFormat.JSON);
+        config.setTimeout(timeout);
+    }
+
+    public void init() throws MojoFailureException {
+
+        try {
+            client = KieServicesFactory.newKieServicesClient(config);
+        } catch (RuntimeException kieEx) {
+            throw new MojoFailureException("error on enstablish connection with remote server: " + kieEx.getMessage());
+        }
+
+    }
+
+    public void deploy(MavenProject project, String container) throws MojoFailureException {
+
+        KieContainerResource kieContainer = new KieContainerResource();
+        kieContainer.setContainerId(container);
+
+        ReleaseId releaseId = new ReleaseId();
+        releaseId.setGroupId(project.getGroupId());
+        releaseId.setArtifactId(project.getArtifactId());
+        releaseId.setVersion(project.getVersion());
+        kieContainer.setReleaseId(releaseId);
+
+        KieServerConfigItem configItem = new KieServerConfigItem();
+        configItem.setName("RuntimeStrategy");
+        configItem.setValue("PER_PROCESS_INSTANCE");
+        configItem.setType("BPM");
+        kieContainer.addConfigItem(configItem);
+
+        ServiceResponse<KieContainerResource> serviceResponse = client.createContainer(container, kieContainer);
+        ServiceResponse.ResponseType type = serviceResponse.getType();
+        String msg = serviceResponse.getMsg();
+
+        LOG.info("Deploy Response: {}", serviceResponse);
+
+        if (ServiceResponse.ResponseType.FAILURE.equals(type)) {
+            throw new MojoFailureException(msg);
+        }
+
+    }
+
+    public void dispose(String container) throws MojoFailureException {
+
+        ServiceResponse<Void> serviceResponse = client.disposeContainer(container);
+        ServiceResponse.ResponseType type = serviceResponse.getType();
+        String msg = serviceResponse.getMsg();
+
+        LOG.info("Dispose Response: {}", serviceResponse);
+
+        if (ServiceResponse.ResponseType.FAILURE.equals(type)) {
+            throw new MojoFailureException(msg);
+        }
+
+    }
+
+    public void replace(MavenProject project, String container) throws MojoFailureException {
```

Comment:
I would actually call it update instead of replace as it's not really replace...